dd-trace-ot/src/main/java/datadog/trace/common/writer/ZipkinV2Api.java

```diff
@@ -67,22 +66,24 @@ public Response sendTraces(final List<List<DDSpan>> traces) {
   }
 
   @Override
-  public byte[] serializeTrace(final List<DDSpan> trace) throws IOException {
-    final ByteArrayOutputStream stream = new ByteArrayOutputStream(trace.size() * 128);
+  public SerializedBuffer serializeTrace(final List<DDSpan> trace) throws IOException {
+    final StreamingSerializedBuffer stream = new StreamingSerializedBuffer(trace.size() * 128);
     final JsonGenerator jsonGenerator = JSON_FACTORY.createGenerator(stream, JsonEncoding.UTF8);
 
     jsonGenerator.writeStartArray();
-    for (DDSpan span : trace) {
-      writeSpan(span, jsonGenerator);
+    for (int i = 0; i < trace.size(); i++) {
```

Comment:
Is this iterator adding overhead?

Reply:
It's a tiny piece of the overall puzzle and almost certainly elided by escape analysis, but every bit matters for this particular allocation hot path.